service/worker/deployment/deployment_util.go

```diff
@@ -91,8 +96,7 @@ func ValidateDeploymentWfParams(fieldName string, field string, maxIDLengthLimit
 // EscapeChar is a helper which escapes the DeploymentWorkflowIDDelimeter character
 // in the input string
 func escapeChar(s string) string {
-	s = strings.Replace(s, `\`, `\\`, -1)
-	s = strings.Replace(s, DeploymentWorkflowIDDelimeter, `\`+DeploymentWorkflowIDDelimeter, -1)
+	s = strings.Replace(s, DeploymentWorkflowIDDelimeter, `|`+DeploymentWorkflowIDDelimeter, -1)
```

Comment:
this is still not injective, `GenerateDeploymentWorkflowID("abc:|","|def")` and `GenerateDeploymentWorkflowID("abc|",":|def")` generate the same thing (though that was trickier to find).

I'll fix it in my PR